contract_mandate_default/hooks.py

```diff
@@ -0,0 +1,19 @@
+# Copyright NuoBiT Solutions, S.L. (<https://www.nuobit.com>)
+# Frank Cespedes <fcespedes@nuobit.com>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl)
+
+from odoo import SUPERUSER_ID, api, fields
+
+
+def post_init_hook(cr, registry):
+    env = api.Environment(cr, SUPERUSER_ID, {})
+    contracts = env["contract.contract"].search(
+        [
+            ("payment_mode_id.payment_method_id.mandate_required", "=", True),
+            ("mandate_id", "!=", False),
+        ]
+    )
+
+    for contract in contracts:
+        if not contract.date_end or contract.date_end > fields.Date.today():
```

Comment:
Why don't you put this inside the previous `search`? `date_end` is a stored attribute.